File: scripts/update-contemporary-images.js
#!/usr/bin/env node
/**
 * Update Contemporary artworks with image URLs
 *
 * Usage:
 * 1. Add your images to public/images/contemporary/
 * 2. Name them using accession numbers (e.g., 2013.443A-E.jpg, 1989.35.png)
 * 3. Run: node scripts/update-contemporary-images.js
 */

const fs = require('fs');
const path = require('path');
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config({ path: '.env.local' });

const supabase = createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL,
  process.env.SUPABASE_SERVICE_ROLE_KEY
);

async function updateContemporaryImages() {
  console.log('='.repeat(80));
  console.log('UPDATING CONTEMPORARY ARTWORK IMAGES');
  console.log('='.repeat(80));

  // Get all Contemporary artworks
  const { data: artworks, error } = await supabase
    .table('Artworks')
    .select('ID, "Accession Number", Title, "Image URL"')
    .eq('Collection', 'Contemporary')
    .order('ID');

  if (error) {
    console.error('Error fetching artworks:', error);
    return;
  }

  console.log(`\nFound ${artworks.length} Contemporary artworks`);

  // Check what images exist in the folder
  const imagesDir = path.join(process.cwd(), 'public', 'images', 'contemporary');

  if (!fs.existsSync(imagesDir)) {
    console.error(`\nError: Images directory does not exist: ${imagesDir}`);
    console.log('Please create it first: mkdir -p public/images/contemporary');
    return;
  }

  const imageFiles = fs.readdirSync(imagesDir)
    .filter(f => /\.(jpg|jpeg|png|gif|webp)$/i.test(f));

  console.log(`\nFound ${imageFiles.length} image files in public/images/contemporary/`);

  if (imageFiles.length === 0) {
    console.log('\nNo images found. Add your images to public/images/contemporary/');
    console.log('Name them using accession numbers, e.g.:');
    console.log('  - 2013.443A-E.jpg');
    console.log('  - 1989.35.png');
    console.log('  - 69.36.7.jpg');
    return;
  }

  console.log('\nImages found:');
  imageFiles.forEach(f => console.log(`  - ${f}`));

  // Create a map of accession number to image file
  const imageMap = {};
  imageFiles.forEach(filename => {
    // Extract accession number from filename (before the extension)
    const basename = filename.replace(/\.(jpg|jpeg|png|gif|webp)$/i, '');
    // Try to normalize accession format
    const normalized = basename.toUpperCase();
    imageMap[normalized] = filename;
    // Also store original case
    imageMap[basename] = filename;
  });

  console.log('\n' + '='.repeat(80));
  console.log('MATCHING ARTWORKS TO IMAGES');
  console.log('='.repeat(80));

  let updated = 0;
  let skipped = 0;
  let alreadyHasImage = 0;

  for (const artwork of artworks) {
    const accession = artwork['Accession Number'];

    // Skip if already has an image URL
    if (artwork['Image URL']) {
      alreadyHasImage++;
      continue;
    }

    // Try to find matching image
    const imageFile = imageMap[accession] ||
                      imageMap[accession.toUpperCase()] ||
                      imageMap[accession.toLowerCase()];

    if (imageFile) {
      const imageUrl = `/images/contemporary/${imageFile}`;

      // Update database
      const { error: updateError } = await supabase
        .table('Artworks')
        .update({ 'Image URL': imageUrl })
        .eq('ID', artwork.ID);

      if (updateError) {
        console.log(`✗ Failed ${artwork.ID} (${accession}): ${updateError.message}`);
        skipped++;
      } else {
        console.log(`✓ Updated ${artwork.ID} (${accession}): ${artwork.Title} → ${imageUrl}`);
        updated++;
      }
    } else {
      console.log(`  Skipped ${artwork.ID} (${accession}): No matching image file`);
      skipped++;
    }
  }

  console.log('\n' + '='.repeat(80));
  console.log('SUMMARY');
  console.log('='.repeat(80));
  console.log(`✓ Updated: ${updated} artworks`);
  console.log(`  Already had images: ${alreadyHasImage} artworks`);
  console.log(`  Skipped (no match): ${skipped} artworks`);
  console.log(`  Total Contemporary: ${artworks.length} artworks`);

  if (skipped > 0) {
    console.log('\nArtworks without images:');
    for (const artwork of artworks) {
      const accession = artwork['Accession Number'];
      const imageFile = imageMap[accession] || imageMap[accession.toUpperCase()];
      if (!imageFile && !artwork['Image URL']) {
        console.log(`  - ${accession}: ${artwork.Title}`);
        console.log(`    Expected filename: ${accession}.jpg or ${accession}.png`);
      }
    }
  }
}

updateContemporaryImages().catch(console.error);

File: scripts/update-contemporary-images.py
#!/usr/bin/env python3
"""
Update Contemporary artworks with image URLs
"""
import os
from supabase import create_client

SUPABASE_URL = "https://wyanldczdzjmmqjtobcv.supabase.co"
SUPABASE_KEY = "sb_secret_KfMFAgEr_9kcdhVngyUvWA_V3MzjGqR"

supabase = create_client(SUPABASE_URL, SUPABASE_KEY)

print("=" * 80)
print("UPDATING CONTEMPORARY ARTWORK IMAGES")
print("=" * 80)

# Get all Contemporary artworks
result = supabase.table('Artworks').select(
    'ID, "Accession Number", Title, "Image URL"'
).eq('Collection', 'Contemporary').order('ID').execute()

artworks = result.data
print(f"\nFound {len(artworks)} Contemporary artworks")

# Check what images exist
images_dir = 'public/images/contemporary'
if not os.path.exists(images_dir):
    print(f"Error: Images directory not found: {images_dir}")
    exit(1)

image_files = [f for f in os.listdir(images_dir)
               if f.lower().endswith(('.jpg', '.jpeg', '.png', '.gif', '.webp'))]

print(f"Found {len(image_files)} image files\n")

# Create map of accession -> image file
image_map = {}
for filename in image_files:
    # Extract accession from filename (before extension)
    accession = os.path.splitext(filename)[0]
    image_map[accession] = filename

print("=" * 80)
print("MATCHING AND UPDATING")
print("=" * 80)

updated = 0
skipped = 0
already_has = 0

for artwork in artworks:
    accession = artwork['Accession Number']

    # Skip if already has image
    if artwork.get('Image URL'):
        already_has += 1
        continue

    # Look for matching image
    if accession in image_map:
        image_file = image_map[accession]
        image_url = f"/images/contemporary/{image_file}"

        # Update database
        try:
            supabase.table('Artworks').update({
                'Image URL': image_url
            }).eq('ID', artwork['ID']).execute()

            print(f"OK {artwork['ID']:3d} | {accession:15s} | {artwork['Title'][:50]:50s} -> {image_file}")
            updated += 1
        except Exception as e:
            print(f"X Failed {artwork['ID']} ({accession}): {e}")
            skipped += 1
    else:
        print(f"  Skipped {artwork['ID']} ({accession}): No matching image")
        skipped += 1

print("\n" + "=" * 80)
print("SUMMARY")
print("=" * 80)
print(f"OK Updated: {updated} artworks")
print(f"  Already had images: {already_has}")
print(f"  Skipped (no match): {skipped}")
print(f"  Total: {len(artworks)} artworks")

if skipped > 0:
    print("\nArtworks without images:")
    for artwork in artworks:
        accession = artwork['Accession Number']
        if accession not in image_map and not artwork.get('Image URL'):
            print(f"  - {accession}: {artwork['Title']}")
